### ISSUE TITLE
Cannot find source file: src/exception.h

### ISSUE THREAD

**Issue by ttpro1995** [CopernicaMarketingSoftware/AMQP-CPP#123]
I run cmake .

```

-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Warning (dev) at CMakeLists.txt:52 (set):
  Cannot set "AMQP-CPP_INCLUDE_PATH": current scope has no parent.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
CMake Error at CMakeLists.txt:36 (add_library):
  Cannot find source file:

    src/exception.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


-- Build files have been written to: /home/ubuntu/workspace/AMQP-CPP
```

**Comment by gatkinso** [comment]
I can confirm this issue.

**Comment by ilovezfs** [comment]
@RafalGoslawski I'm seeing this while attempting to build the 2.7.0 upgrade for Homebrew in https://github.com/Homebrew/homebrew-core/pull/13000

The following patch seems to help:
```
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d52cbd9..3d01c83 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -46,7 +46,7 @@ deferredcancel.cpp
 deferredconsumer.cpp
 deferredconsumerbase.cpp
 deferredget.cpp
-exception.h
+../include/exception.h
 exchangebindframe.h
 exchangebindokframe.h
 exchangedeclareframe.h
@@ -59,14 +59,13 @@ exchangeunbindokframe.h
 extframe.h
 field.cpp
 flags.cpp
-frame.h
+../include/frame.h
 framecheck.h
 headerframe.h
 heartbeatframe.h
 includes.h
-messageimpl.h
 methodframe.h
-protocolexception.h
+../include/protocolexception.h
 protocolheaderframe.h
 queuebindframe.h
 queuebindokframe.h
```
In addition to that I need to `cp include/endian.h src/` and then it builds.

**Comment by ilovezfs** [comment]
@RafalGoslawski `endian.h` is still causing

```
[ 21%] Building CXX object CMakeFiles/amqp-cpp.dir/src/connectionimpl.cpp.o
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -Damqp_cpp_EXPORTS -I/tmp/amqp-cpp-20170504-94143-xbosvw/AMQP-CPP-2.7.1  -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -fPIC   -std=c++11 -o CMakeFiles/amqp-cpp.dir/src/connectionimpl.cpp.o -c /tmp/amqp-cpp-20170504-94143-xbosvw/AMQP-CPP-2.7.1/src/connectionimpl.cpp
In file included from /tmp/amqp-cpp-20170504-94143-xbosvw/AMQP-CPP-2.7.1/src/connectionimpl.cpp:13:
/tmp/amqp-cpp-20170504-94143-xbosvw/AMQP-CPP-2.7.1/src/passthroughbuffer.h:20:10: fatal error: 'endian.h' file not found
#include "endian.h"
         ^
1 error generated.
make[2]: *** [CMakeFiles/amqp-cpp.dir/src/connectionimpl.cpp.o] Error 1
make[1]: *** [CMakeFiles/amqp-cpp.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by ilovezfs** [comment]
@RafalGoslawski it looks like that include statement just needs to be removed from passthroughbuffer.h
